browser/ui/webui/webcompat_reporter/webcompat_reporter_ui.cc

```diff
@@ -0,0 +1,178 @@
+#include "brave/components/brave_shields/common/pref_names.h"
+#include "brave/components/brave_vpn/common/buildflags/buildflags.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/webcompat_reporter/browser/fields.h"
+#include "brave/components/webcompat_reporter/browser/webcompat_report_uploader.h"
+#include "brave/components/webcompat_reporter/resources/grit/webcompat_reporter_generated_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/language/core/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/browser/web_ui_message_handler.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "ui/web_dialogs/web_dialog_delegate.h"
+#include "url/gurl.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/browser/brave_vpn/brave_vpn_service_factory.h"
+#include "brave/components/brave_vpn/browser/brave_vpn_service.h"
+#endif
+
+namespace webcompat_reporter {
+
+namespace {
+
+class WebcompatReporterDOMHandler : public content::WebUIMessageHandler {
+ public:
+  explicit WebcompatReporterDOMHandler(Profile* profile);
+  WebcompatReporterDOMHandler(const WebcompatReporterDOMHandler&) = delete;
+  WebcompatReporterDOMHandler& operator=(const WebcompatReporterDOMHandler&) =
+      delete;
+  ~WebcompatReporterDOMHandler() override;
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  void InitAdditionalParameters(Profile* profile);
+  void HandleSubmitReport(const base::Value::List& args);
+
+  std::unique_ptr<webcompat_reporter::WebcompatReportUploader> uploader_;
+  std::string ad_block_list_names_;
+  std::string languages_;
+  bool language_farbling_enabled_ = true;
+  bool brave_vpn_connected_ = false;
+};
+
+WebcompatReporterDOMHandler::WebcompatReporterDOMHandler(Profile* profile)
+    : uploader_(std::make_unique<webcompat_reporter::WebcompatReportUploader>(
+          profile->GetURLLoaderFactory())) {
+  InitAdditionalParameters(profile);
+}
+
+void WebcompatReporterDOMHandler::InitAdditionalParameters(Profile* profile) {
+  std::vector<std::string> ad_block_list_names;
+
+  // Collect all enabled adblock list names
+  brave_shields::AdBlockService* ad_block_service =
+      g_brave_browser_process->ad_block_service();
+  if (ad_block_service != nullptr) {
+    brave_shields::AdBlockRegionalServiceManager* regional_service_manager =
+        ad_block_service->regional_service_manager();
+    CHECK(regional_service_manager);
+    for (const brave_shields::FilterListCatalogEntry& entry :
+         regional_service_manager->GetFilterListCatalog()) {
+      if (regional_service_manager->IsFilterListEnabled(entry.uuid)) {
+        ad_block_list_names.push_back(entry.title);
+      }
+    }
+  }
+
+  ad_block_list_names_ = base::JoinString(ad_block_list_names, ",");
```

Comment:
I'd personally recommend using component IDs here for slightly easier machine-readability/consistency with the catalog in https://github.com/brave/adblock-resources (list names may change, infrequently). Then again, I'm not going to be the one looking at the raw report uploads :smile:

Reply:
happy to change this if it would be more useful. as you know, all of our webcompat reports are displayed on our dashboard in a human readable format, for manual review. i assumed it would be more useful for an adblock/easylist dev to see the name of the list in this case. but please correct me if i'm wrong since I will also not be looking at the reports! :smile: . curious to hear your thoughts on my perspective

Reply:
I don't feel strongly about it. We'd be able to figure out what lists are referenced either way if reviewing raw upload data, and having the names directly available makes it easier to do so without any changes to the tooling.

The benefits of using the component IDs are
1. fixed size/format
2. constant identifier even if list names change

(1) is 100% a serverside implementation concern, so I have no preference there
(2) is slightly nicer for the sake of consistency in longer-term analysis, but we delete reports after 30 days anyways so it's not even too relevant either

Reply:
Thanks for elaborating. If using names would make report reviews easier, then I would prefer to go that route. From a server-side perspective, it's much easier for us to store the names, as we will not have to maintain a mapping of component IDs to list names. The size of the field is not a concern, considering the low volume of reports (compared to the rest of the stats data we manage). If you think that potential name changes won't cause too much trouble, then I hope to keep this as is